bin/plot_truncated_pexecs

```diff
@@ -0,0 +1,221 @@
+                trials[n_pexecs][key] = list()
+                for _ in xrange(NUMBER_TRIALS):
+                    counts = [0, 0, 0, 0]
+                    for _ in xrange(n_pexecs):
+                        next_class = rangen.choice(classifications[key])
+                        counts[CATEGORIES.index(next_class)] += 1
+                    trials[n_pexecs][key].append(counts)
+    return classifications, trials
+
+
+def do_cis_differ((x1, y1), (x2, y2)):
+    """Given to CIs return True if they do NOT overlap."""
+
+    return y1 < x2 or y2 < x1
+
+
+def count_changed_trials(classifications, trials):
+    """What proportion of trials changed classifications?"""
+
+    num_similar = dict()  # n_pexecs -> int
+    num_different = dict()  # n_pexecs -> int
+    # Generate confidence intervals for the data in the original experiment.
+    original_cis = dict()
+    for key in classifications:
+        original_cis[key] = dict()
+        for category in CATEGORIES:
+            original_cis[key][category] = \
+                proportion_confint(classifications[key].count(category), ORIGINAL_PEXECS,
+                                   alpha=ALPHA, method=METHOD)
+    # Compare to data from the Bernoulli trials.
+    for n_pexecs in xrange(MIN_PEXECS, MAX_PEXECS + 1):
+        num_similar[n_pexecs] = 0
+        num_different[n_pexecs] = 0
+        for key in classifications:
+            if all_same_category(classifications[key]):
+                # If the original pexecs for this key all had same classification,
+                # no pexecs will have changed in the trials.
+                num_similar[n_pexecs] += NUMBER_TRIALS
+            else:
+                # Of the NUMBER_TRIALS we generated for this n_pexecs, how many
+                # trials resulted in a significantly different set of classifications
+                # to the original? Here be dragons.
+                for trial in xrange(NUMBER_TRIALS):
+                    similar = True  # Assume trial is similar to original.
+                    for category in CATEGORIES:
+                        count = trials[n_pexecs][key][trial].count(category)
+                        this_ci = proportion_confint(count, n_pexecs, alpha=ALPHA, method=METHOD)
+                        if do_cis_differ(original_cis[key][category], this_ci):
+                            similar = False
+                            break
+                    if similar:
+                        num_similar[n_pexecs] += 1
+                    else:
+                        num_different[n_pexecs] += 1
+    return num_similar, num_different
+
+
+def draw_plot(same, changed, outfile):
+    """Plot 'same' data and write to PDF file."""
+
+    pdf = PdfPages(outfile)
+    fig, axis = pyplot.subplots()
+    # Prepare data.
+    assert same.keys() == changed.keys()
+    x_vals = sorted(same.keys())
+    y_vals = [float(same[x]) / (changed[x] + same[x]) * 100.0 for x in sorted(same.keys())]
```

Comment:
This is the core calculation. What happens if `same` and `changed` (should that be `different`?) have been inverted somewhere. If this was:

```python
y_vals = [float(changed[x]) / (changed[x] + same[x]) * 100.0 for x in sorted(same.keys())]
```

I think we'd get the shape of graph we expect. Of course, that suggests there's a bug elsewhere...